bin/network-monitor/src/deploy/counter.rs

```diff
@@ -0,0 +1,72 @@
+//! Counter program account creation functionality.
+
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use miden_lib::transaction::TransactionKernel;
+use miden_objects::account::{
+    Account,
+    AccountBuilder,
+    AccountComponent,
+    AccountFile,
+    AccountId,
+    AccountStorageMode,
+    AccountType,
+    StorageSlot,
+};
+use miden_objects::{Felt, Word};
+use tracing::instrument;
+
+use crate::COMPONENT;
+
+/// Create a counter program account with custom MASM script.
+#[instrument(target = COMPONENT, name = "create-counter-account", skip_all, ret(level = "debug"))]
+pub fn create_counter_account(owner_account_id: AccountId) -> Result<Account> {
+    // Load and customize the MASM script
+    let script =
+        include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/assets/counter_program.masm"));
+
+    // Compile the account code
+    let owner_felts: [Felt; 2] = owner_account_id.into();
+    let owner_word = Word::from([owner_felts[0], owner_felts[1], Felt::new(0), Felt::new(0)]);
```

Comment:
Based how these will be used, would it be better to arrange them differently?


```suggestion
    let owner_word = Word::from([Felt::ZERO, Felt::ZERO, owner_felts[0], owner_felts[1]]);
```